crypto/modes/siv128.c

```diff
@@ -0,0 +1,365 @@
+/*
+ * Copyright 2017 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h>
+#include <stdlib.h>
+#include <openssl/crypto.h>
+#include <openssl/cmac.h>
+#include "modes_lcl.h"
+
+#ifndef OPENSSL_NO_SIV
+
+# if !defined(BSWAP8)
+#  if defined(__GNUC__) || defined(__clang__)
+__owur static ossl_inline uint64_t BSWAP8(uint64_t x)
+{
+    return __builtin_bswap64(x);
```

Comment:
I don't see no __builtin_bswap64 in gcc 3, so that check for __GNUC__ along is not sufficient.